app/controllers/api_controller/providers.rb

```diff
@@ -132,13 +137,28 @@ def validate_credential_attributes(provider, creds)
                                     invalid_attrs.join(', '), provider.class.name, auth_attrs.join(", "))
     end
 
+    def check_endpoints(data)
+      connection_configurations = data["connection_configurations"]
```

Comment:
I think you can drop this variable and work with 'data["connection_configurations"]'.